### ISSUE TITLE
Unable to locate "vcvarsall.bat" during Visual Studio 2017 builds

### ISSUE THREAD

**Issue by girardc79** [conan-io/conan-package-tools#54]
When attempting to build with Visual Studio 2017, the following error is produced:

`The system cannot find the path specified.`
`Error while executing:`
`call "%vs150comntools%../../VC/Auxiliary/Build/vcvarsall.bat" amd64 && conan test_package . -s arch="x86_64" -s build_type="Release" -s compiler="Visual Studio" -s compiler.runtime="MT" -s compiler.version="15" -o thelib:shared="False" `

I noted that there was a commit to add support for it:
https://github.com/conan-io/conan-package-tools/commit/c4cace87cf02cec31fccb8ac5d0c3e68b1c1a3f4

But Microsoft has removed the %vsXXXcomntools% variables starting with VS2017 and forward:
https://blogs.msdn.microsoft.com/vcblog/2017/03/06/finding-the-visual-c-compiler-tools-in-visual-studio-2017/

Currently we are working around this by manually creating a vs150comntools variable.

**Comment by memsharded** [comment]
Yes, that is right. Msft has removed the env-var, and conan still needs it to be defined. And it seems that it will keep necessary, because the removal of the env-var is because now different installations of VS can co-exists. Even if they can be somewhat detected (IMHO seems too complex), still just one installation has to be used to build the conan packages, so the env var is still necessary to tell conan which one.

**Comment by 9409215070** [comment]
how to solve?

**Comment by uilianries** [comment]
https://docs.conan.io/en/latest/reference/tools.html#tools-vswhere
